### ISSUE TITLE
User without email and username issue in dj-rest-auth, allauth and react native app

### ISSUE THREAD

**Issue by Jayesh-Mahato** [iMerica/dj-rest-auth#271]
I am using Social Authentication (Facebook, Twitter, Google, Github) for SignUp and Login using React Native Expo, dj-rest-auth, django allauth. In my case, user is getting registered when passing access token through Facebook API endpoint but with no username and email. The user is getting registered in django admin as well with no username and email. Both are simply blank. The following is also added to the django settings:

```
ACCOUNT_EMAIL_REQUIRED = True
SOCIALACCOUNT_EMAIL_REQUIRED = ACCOUNT_EMAIL_REQUIRED
REST_AUTH_REGISTER_SERIALIZERS = {
        'REGISTER_SERIALIZER': 'path.to.OverRegisterSerializer',
}
```

Since Facebook Graph has removed username, I had further implemented the following class to add username based on email, however the email also is blank in my case.

```
from dj_rest_auth.registration.serializers import RegisterSerializer
from rest_framework import serializers
from allauth.utils import get_username_max_length
from allauth.account import app_settings as allauth_settings


class CustomRegistrationSerializer(RegisterSerializer):

  username = serializers.CharField(
        max_length=get_username_max_length(),
        min_length=allauth_settings.USERNAME_MIN_LENGTH,
        required=allauth_settings.USERNAME_REQUIRED,
    )
  email = serializers.EmailField(required=allauth_settings.EMAIL_REQUIRED)

  def validate_username(self, username):
  	username = get_adapter().clean_username(username)
  	if not username:
  		username = email.split('@')[0] + "_" + email.split('@')[1].split('.')[0]
  	return username

  def save(self, request):
        adapter = get_adapter()
        user = adapter.new_user(request)
        self.cleaned_data = self.get_cleaned_data()
        adapter.save_user(request, user, self)
        self.custom_signup(request, user)
        setup_user_email(request, user, [])
        return user

```
My fetch call is listed below:

```
fetch(
    "http://192.168.1.102:8080/dj-rest-auth/facebook/",
    {
      method: "POST",
      headers: {
        'Accept': 'application/json',
        'Content-Type':'application/json',
      },
      xsrfCookieName:"csrftoken",
      xsrfHeaderName:'X-CSRFToken',
      body:JSON.stringify({access_token:resolvedToken})
    }
  )
  .then(resp => resp.json())
  .then(data => {
    console.log(data);
  }  
  )
  .catch(error => console.log(error))
```

I need to have the email address and username for a newly social authenticated user. How can I have it enabled?

**Comment by cmraible** [comment]
Social Logins / Registrations through Facebook/Google etc. don't use the default RegisterSerializer — the default RegisterSerializer, which you are overriding, is only when users manually signup with username/password. I'm not aware of a way to override the SocialLoginSerializer except by overriding the serializer on each individual SocialLoginView (you should have one for each provider you use already)

**Comment by Jayesh-Mahato** [comment]
@cmraible  I am kind of stuck here with overriding on individual SocialLoginView currently. Thanks for your inputs though.

**Comment by Jayesh-Mahato** [comment]
I tried overriding SocialLoginSerializer from dj_rest_auth.registration.serializers and tested for Facebook Login (For twitter TwitterLoginSerializer would need an override - not tried completely yet).
I have added an email field `e_mail = serializers.EmailField(required=False)` and used it for further manipulation of saving both email and username, for Facebook login.

It worked. Will proceed for other social logins now.

@cmraible Thanks for the suggestion.
